src/main/java/org/swisspush/reststorage/RestStorageHandler.java

```diff
@@ -414,63 +416,77 @@ private void putResource(RoutingContext ctx) {
                     message = resource.errorMessage;
                 }
                 ctx.response().end(message);
-                return;
             }
-
-            if (resource.rejected) {
+            else if (resource.rejected) {
                 ctx.response().setStatusCode(StatusCode.CONFLICT.getStatusCode());
                 ctx.response().setStatusMessage(StatusCode.CONFLICT.getStatusMessage());
                 ctx.response().end();
-                return;
             }
-            if (!resource.modified) {
+            else if (!resource.modified) {
                 ctx.response().setStatusCode(StatusCode.NOT_MODIFIED.getStatusCode());
                 ctx.response().setStatusMessage(StatusCode.NOT_MODIFIED.getStatusMessage());
                 ctx.response().headers().set(ETAG_HEADER.getName(), etag);
                 ctx.response().headers().add(CONTENT_LENGTH.getName(), "0");
                 ctx.response().end();
-                return;
             }
-            if (!resource.exists && resource instanceof DocumentResource) {
```

Comment:
I didn't remove that check. I only moved it:

First we check `if (resource instanceof DocumentResource)` at [RestStorageHandler.java line 438](https://github.com/swisspush/vertx-rest-storage/pull/73/files/b04d8c2ff9a2902d657f932f090ced27c5745883#diff-55dc5c01edfbfdccb7489ca50992cc97R438).

Then we also check `if (!resource.exists)` at [RestStorageHandler.java line 455](https://github.com/swisspush/vertx-rest-storage/pull/73/files/b04d8c2ff9a2902d657f932f090ced27c5745883#diff-55dc5c01edfbfdccb7489ca50992cc97R455).

But I think the 2nd check is in the wrong place and I think it would be better to place it BEFORE we're calling method `putResource_handleDocumentResource` this way, all the checks will remain together (Will do).